hbase-server/src/main/java/org/apache/hadoop/hbase/ipc/CallRunner.java

```diff
@@ -88,6 +88,7 @@ public void setStatus(MonitoredRPCHandler status) {
    * Cleanup after ourselves... let go of references.
    */
   private void cleanup() {
+    this.call.cleanup();
```

Comment:
So this is for safety. And I checked the code, drop is only called in AdaptiveLifoCoDelCallQueue, maybe this is why we do not catch this problem? Can we add a UT for this?

Reply:
if (call.disconnectSince() >= 0) {
        if (RpcServer.LOG.isDebugEnabled()) {
          RpcServer.LOG.debug(Thread.currentThread().getName() + ": skipped " + call);
        }
        return;
      }
      call.setStartTime(System.currentTimeMillis());
      if (call.getStartTime() > call.getDeadline()) {
        RpcServer.LOG.warn("Dropping timed out call: " + call);
        return;
      }

When this two case, the call.cleanup() also not called before.

Reply:
Yes, I got them after seeing your UT.